### DISCUSSION TITLE
Is there any way to embed rclone within the rustic binary?

### DISCUSSION THREAD

**Discussion by nickchomey** [rustic-rs/rustic#999] [Ideas]
I'm wondering if there's any way to embed rclone within rustic so that only a single binary is needed?

I dont know anything about rust, but I see that there is [librclone](https://github.com/rclone/rclone/tree/master/librclone) and a [rust crate](https://crates.io/crates/librclone) for it

I suppose that even if it is possible, perhaps it would be preferrable not to - so that rclone can update independently (and seemingly more frequently) from rustic

**Comment by simonsan** [comment]
Hey there, corresponding issue is here: https://github.com/rustic-rs/rustic_core/issues/14

We are currently integrating with openDAL in https://github.com/rustic-rs/rustic_core/pull/73 

So in general, I'm not sure if embedding rclone is the way to go, also because it's not supported on Windows and might create a patchwork rug within rustics codebase to keep it cross-platform.

**Reply by nickchomey** [reply]
Ah, very cool. Thanks for the information. I look forward to seeing how those turn out!
